espflash/src/cli/mod.rs

```diff
@@ -214,35 +207,43 @@ pub struct ReadFlashArgs {
 #[non_exhaustive]
 #[group(skip)]
 pub struct SaveImageArgs {
-    /// Custom bootloader for merging
-    #[arg(long, value_name = "FILE")]
-    pub bootloader: Option<PathBuf>,
     /// Chip to create an image for
     #[arg(long, value_enum)]
     pub chip: Chip,
     /// File name to save the generated image to
     pub file: PathBuf,
-    /// Minimum chip revision supported by image, in format: major.minor
-    #[arg(long, default_value = "0.0", value_parser = parse_chip_rev)]
-    pub min_chip_rev: u16,
     /// Boolean flag to merge binaries into single binary
     #[arg(long)]
     pub merge: bool,
-    /// Custom partition table for merging
-    #[arg(long, short = 'T', requires = "merge", value_name = "FILE")]
```

Comment:
This change does remove the `requires = "merge"` setting from the `partition_table` argument for `save-image`, since the `ImageArgs` can't see the `merge` flag.  This attribute was previously different between the `save-image` and `flash` commands.

We could add back our own manual check to reject the `--partition-table` flag if `--merge` is not specified.  However, it does look like the `partition_table` argument is in fact used to print the partition size even when the `--merge` flag is not specified.